### ISSUE TITLE
R treating relogit model as lm instead of glm?

### ISSUE THREAD

**Issue by ppdebem** [IQSS/Zelig#291]
I extracted the fitted model from a Zelig (relogit) object with the `from_zelig_model` function and as i try to use it to predict on a new dataset i keep getting this warning:
```
Warning messages:
1: In predict.lm(object, newdata, se.fit, scale = 1, type = ifelse(type ==  :
  calling predict.lm(<fake-lm-object>) ...
```
The prediction goes through despite the warning, but is that alright?

**Comment by christophergandrud** [comment]
Hi @ppdebem. Thanks for your query. 

I'm unable to reproduce the behaviour you mention using the documented example:

```R
library(Zelig)
data(mid)
z_out <- zelig(conflict ~ major + contig + power + maxdem + mindem + years,
                data = mid, model = "relogit", tau = 1042/303772, cite = FALSE)
model_out <- from_zelig_model(z_out)
class(model_out)

## [1] "Relogit" "glm" 
```

`predict` doesn't return an error when used with this `model_out`.

Could you provide a minimal reproducible example we could look into?

**Comment by ppdebem** [comment]
Something must be wrong with my side of things then, if i call the predict function on the model generated by your example it also throws the warning:

`p <- predict(model_out, mid, type="response")`
`Warning message:
In predict.lm(object, newdata, se.fit, scale = 1, type = ifelse(type ==  :
  calling predict.lm(<fake-lm-object>) ...`

...and the prediction still goes through anyway:

`head(p)`

I'll try to look into it some more on my side.

**Comment by christophergandrud** [comment]
Hm, that is strange. Have you terminated your R session and restarted (maybe there is some sort of package conflict)?

**Comment by ppdebem** [comment]
Starting a clean session didn't seem to work. I tried it in a fresh R install in another computer and the same keeps happening. Here is my R version info and loaded packages:
```R
> version
               _                           
platform       x86_64-w64-mingw32          
arch           x86_64                      
os             mingw32                     
system         x86_64, mingw32             
status                                     
major          3                           
minor          4.1                         
year           2017                        
month          06                          
day            30                          
svn rev        72865                       
language       R                           
version.string R version 3.4.1 (2017-06-30)
nickname       Single Candle               
> (.packages())
[1] "stats"     "graphics"  "grDevices" "utils"     "datasets"  "methods"  
[7] "base"     
```
Not sure what could be the problem, i'll keep looking.

**Comment by christophergandrud** [comment]
What version of Zelig are you using?

**Comment by ppdebem** [comment]
I was using version 5.1-2 straight from CRAN. I just tried it with the github version (5.1-2.9000) but its still doing the same.

**Comment by christophergandrud** [comment]
Hm, this seems like a really difficult issue to diagnose if I can't reproduce the problem. Have you ever had a period where you were able to use `predict` with `relogit`?

**Comment by ppdebem** [comment]
Unfortunately it was like this since the very first time i tried it.

I decided to try it in a linux install (ubuntu 17.04) thinking it might be related to my windows/R installations but i still got the warning using my data and same example from above.

**Comment by izahn** [comment]
Here is a reproducible example:
```
library(Zelig)
x <- Zelig:::relogit(conflict ~ major, data = mid, tau = 1042/303772)
foo <- predict(x, newdata = mid)
```
Note that 
```
foo <- predict(x)
```
does *not* produce the error (refer to `predict.glm` to see why).

**Comment by christophergandrud** [comment]
Thanks @izahn. Simple fix in 1535f43c71e061505d1052e1047ca69683fd6190
